vjunosswitch/docker/launch.py

```diff
@@ -0,0 +1,160 @@
+
+def handle_SIGTERM(signal, frame):
+    sys.exit(0)
+
+signal.signal(signal.SIGINT, handle_SIGTERM)
+signal.signal(signal.SIGTERM, handle_SIGTERM)
+signal.signal(signal.SIGCHLD, handle_SIGCHLD)
+
+TRACE_LEVEL_NUM = 9
+logging.addLevelName(TRACE_LEVEL_NUM, "TRACE")
+
+def trace(self, message, *args, **kws):
+    # Yes, logger takes its '*args' as 'args'.
+    if self.isEnabledFor(TRACE_LEVEL_NUM):
+        self._log(TRACE_LEVEL_NUM, message, args, **kws)
+logging.Logger.trace = trace
+
+class VJUNOSSWITCH_vm(vrnetlab.VM):
+    def __init__(self, hostname, username, password, conn_mode):
+        for e in os.listdir("/"):
+            if re.search(".qcow2$", e):
+                disk_image = "/" + e
+        super(VJUNOSSWITCH_vm, self).__init__(username, password, disk_image=disk_image, ram=5120)
+
+        # these QEMU cmd line args are translated from the shipped libvirt XML file
+        self.qemu_args.extend(["-smp", "4,sockets=1,cores=4,threads=1"])
+        # Additional CPU info
+        self.qemu_args.extend([
+            "-cpu", "IvyBridge,vme=on,ss=on,vmx=on,f16c=on,rdrand=on,hypervisor=on,arat=on,tsc-adjust=on,umip=on,arch-capabilities=on,pdpe1gb=on,skip-l1dfl-vmentry=on,pschange-mc-no=on,bmi1=off,avx2=off,bmi2=off,erms=off,invpcid=off,rdseed=off,adx=off,smap=off,xsaveopt=off,abm=off,svm=off"
+            ])
+        # mount config disk with juniper.conf base configs 
+        self.qemu_args.extend(["-drive", "if=none,id=config_disk,file=/config.img,format=raw",
+                               "-device", "virtio-blk-pci,drive=config_disk"])
+        self.qemu_args.extend(["-overcommit", "mem-lock=off"])
+        self.qemu_args.extend(["-display", "none", "-no-user-config", "-nodefaults", "-boot", "strict=on"])
+        self.nic_type = "virtio-net-pci"
+        self.num_nics = 11
+        self.hostname = hostname
+        self.smbios = ["type=1,product=VM-VEX"]
+        self.qemu_args.extend(["-machine", "pc-i440fx-focal,usb=off,dump-guest-core=off,accel=kvm"])
+        self.qemu_args.extend(["-device", "piix3-usb-uhci,id=usb,bus=pci.0,addr=0x1.0x2"])
+        self.conn_mode = conn_mode
+
+    def bootstrap_spin(self):
+        """ This function should be called periodically to do work.
+        """
+
+        if self.spins > 300:
+            # too many spins with no result ->  give up
+            self.stop()
+            self.start()
+            return
+
+        # lets wait for the OS/platform log to determine if VM is booted 
+        (ridx, match, res) = self.tn.expect([b"FreeBSD/amd64"], 1)
+        if match: # got a match!
+            if ridx == 0: # login
+                self.logger.info("VM started")
+
+                # Login
+                self.wait_write("\r", None)
+                self.wait_write("admin", wait="login:")
+                self.wait_write(self.password, wait="Password:")
+                self.wait_write("\r", wait="admin@vJunos-switch>")
+                self.logger.info("Login completed")
```

Comment:
@akielaries all our nodes are configured with a hostname mathchig the lab node name.
This is done at the bootstrap stage. Looks like your default config should take this into account 

see https://github.com/hellt/vrnetlab/blob/16b94ab54bade94c00dbd55b7e56f68406723ef7/sros/docker/launch.py#L800

Reply:
This sounds good! I went ahead and changed the `launch.py` script to fill in the host name in the `juniper.conf` file based on the lab node name.